### PR TITLE
[Merge] Implement engine_consensusValidated EE API function

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

Based atop of PR #4396 

Just override the topmost `BlockProcessor.processAndValidateBlock()` method (which is called directly from `ForkChoice.onBlock()`) in the `BlockProcessorMerge` and invoke `engine_consensusValidated()` 
- with a `VALID` result if the `processAndValidateBlock()` completes normally and returns transitioned `BeaconState`
- with an `INVALID` result if a `StateTransitionException` occurs

## Fixed Issue(s)

A part of #4346 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/executionengine/ExecutionEngineChannel.java [hunk 1]
```diff
@@ -25,6 +25,11 @@
 
 public interface ExecutionEngineChannel extends ChannelInterface {
 
+  enum ConsensusValidationResult {
```
Comment (tbenr) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/executionengine/ExecutionEngineChannel.java:34]: I think here we should agree on a coherent naming convention. I used `ExecutionPayloadStatus` for `engine_executePayload`. (status has been inspired by the field in the API response)

File: services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/ExecutionEngineChannelImpl.java [hunk 2]
```diff
@@ -68,34 +66,24 @@ private static void printConsole(String formatString, Object... args) {
   }
 
   @Override
-  public SafeFuture<UInt64> preparePayload(
```
Comment (tbenr) [inline services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/ExecutionEngineChannelImpl.java]: seems like you're rolling back my previous PR :-)

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/executionengine/ExecutionEngineChannel.java (hunk 1 above)
Comment (Nashatyrev) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/executionengine/ExecutionEngineChannel.java:34]: I don't think this is too important. `ConsensusValidationStatus` works for me as well, but for some reason I like it less :)

File: services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/ExecutionEngineChannelImpl.java (hunk 2 above)
Comment (Nashatyrev) [reply services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/ExecutionEngineChannelImpl.java]: Yep, sorry. Sloppy merge :( 
Already fixed

**Review by tbenr — APPROVED** [review]
LGTM a double-check from @mkalinin could be nice

**Review by mkalinin — APPROVED** [review]
LGTM 👍

**Pull request merged**